03c_other_dashboard_elements.ipynb

```diff
@@ -0,0 +1,413 @@
+    "            self.original_data['Temperature'], \n",
+    "            self.window_size, \n",
+    "            self.polynomial_order\n",
+    "        )\n",
+    "        \n",
+    "        self.selected = (\n",
+    "            self.original_data[(self.original_data['Year'] >= self.year_range[0]) & \n",
+    "                               (self.original_data['Year'] <= self.year_range[1])]\n",
+    "        )\n",
+    "        self.display_plot()\n",
+    "        self.display_data()\n",
+    "\n",
+    "    def display_plot(self): \n",
+    "        # This plotting function is copied from the first \n",
+    "        # version of the dashboard.\n",
+    "        self.plot_output.clear_output(wait=True) \n",
+    "        with self.plot_output: \n",
+    "            plt.xlabel('Year') \n",
+    "            plt.ylabel('Temperature Anomalies over Land w.r.t. 1951-80 (˚C)') \n",
+    "            plt.title('Global Annual Mean Surface Air Temperature Change')\n",
+    "            plt.plot(self.selected['Year'], self.selected['Temperature'], label='Raw Data') \n",
+    "            plt.plot(self.selected['Year'], self.selected['Savitzky-Golay'], label='Smoothed Data') \n",
+    "            plt.legend()\n",
+    "            plt.show() \n",
+    "\n",
+    "    def display_data(self):\n",
+    "        self.data_output.data = self.selected"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "67c77c39-7a9d-4009-b6f0-7707bb901c33",
+   "metadata": {},
+   "source": [
+    "Let's make one of these objects and display its `plot_output`, which is an `Output` widget."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1309027c-d2c3-4cac-bc8b-f109b218ddc2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dap = DataAndPlot()\n",
+    "dap.plot_output"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0744b2e7-6ea2-4c8e-a618-307a0a2c8389",
+   "metadata": {},
+   "source": [
+    "Because our class observes the value of `smoothing_info`, setting `smoothing_info` should generate a plot."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5ca33f85-dee4-45f8-9b17-af85b58e6d53",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dap.smoothing_info = dict(year_range=(1950, 2000), window_size=5, polynomial_order=2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "10b69cc9-458f-4b71-8820-cb44f301d60f",
+   "metadata": {},
+   "source": [
+    "## Mutable traits are a little tricky"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "097fd19c-32c3-4d32-8c1d-53938fd34740",
+   "metadata": {},
+   "source": [
+    "One important thing to watch out for that applies to widgets broadly. When the value of a trait is a *mutable* type, i.e. a type whose contents can change, like a `list` or a `dict`, changing the values inside that mutable object will not trigger a change in the event.\n",
```

Comment:
I have used functools.partial to solve this problem in the past and would be interested in adding some more cells about that. that solution was one of the main reasons i was able to build one of my more complex apps, and feels very relevant to our focus.

Reply:
That would be excellent!